plotly/tests/test_core/test_jupyter/lib/tape-wrapper.js

```diff
@@ -0,0 +1,22 @@
+var test = require('tape');
+var xhr = require('xhr');
+
+var cnt = 0;
+var noop = function() {};
+
+var post = function(query, data) {
+    var opts = data ? { body: data } : {};
+    xhr.post('/?' + query + '&' + (cnt++), opts, noop);
+};
+
+var ws = test.createStream();
+
+ws.on('data', function(data) {
+    post('data', data)
```

Comment:
Hack into the [tape](https://github.com/substack/tape) test framework, so that instead of logging to the browser console on assertions it makes requests to the test server.